cmd/download.go

```diff
@@ -176,6 +139,161 @@ func runDownload(cfg config.Config, flags *pflag.FlagSet, args []string) error {
 	return nil
 }
 
+type download struct {
+	// either/or
+	slug, uuid string
+
+	// user config
+	token, apibaseurl, workspace string
+
+	// optional
+	track, team string
+
+	payload *downloadPayload
+}
+
+func newDownload(flags *pflag.FlagSet, usrCfg *viper.Viper) (*download, error) {
```

Comment:
This is currently longer than ideal. I think it makes sense to have the logic to populate the payload in the constructor though. Clients can't do anything with a download without the payload being populated and extracting it to a method doesn't seem appropriate because clients should never need to call it - it just needs to happen on creation.